python/GafferUI/SpreadsheetUI.py

```diff
@@ -1541,9 +1535,54 @@ def __plugPopupMenu( menuDefinition, plugValueWidget ) :
 
 def __createSpreadsheetForNode( node, activeRowNamesConnection, selectorContextVariablePlug, selectorValue ) :
 
-	with Gaffer.UndoScope( node.ancestor( Gaffer.ScriptNode ) ) :
+	with Gaffer.UndoScope( node.scriptNode() ) :
 
 		spreadsheet = Gaffer.Spreadsheet( node.getName() + "Spreadsheet" )
+		__drivePlugFromRowNames( activeRowNamesConnection, selectorContextVariablePlug, selectorValue, spreadsheet )
+		node.parent().addChild( spreadsheet )
+
+	GafferUI.NodeEditor.acquire( spreadsheet )
+
+def __drivePlugFromRowNames( activeRowNamesConnection, selectorContextVariablePlug, selectorValue, spreadsheet ) :
+
+	node = activeRowNamesConnection.node()
+
+	# We defer locking the plug until we know we're going ahead
+	lockSelectorVarPlug = False
+	if selectorValue is None :
+		selectorValue = "${" + selectorContextVariablePlug.getValue() + "}"
+		lockSelectorVarPlug = True
+
+	# Check that the sheet's selector meets requirements before making any changes
+
+	existingSelector = spreadsheet["selector"].getValue()
+	if existingSelector and selectorValue and existingSelector != selectorValue :
+
+		message = "{sheetName}'s selector is set to: '{sheetSelector}'.\n\n" + \
+			"The '{plugName}' plug requires a different selector to work\n" + \
+			"properly. Continuing will reset the selector to '{selector}'."
+
+		confirm = GafferUI.ConfirmationDialogue(
+			"Invalid Selector",
+			message.format(
+				sheetName = spreadsheet.getName(), sheetSelector = existingSelector,
+				plugName = activeRowNamesConnection.getName(), selector = selectorValue
+			),
+			confirmLabel = "Continue"
+		)
+
+		scriptWindow = GafferUI.ScriptWindow.acquire( node.scriptNode() )
```

Comment:
This is a bit awkward given that we now have floating panels and the ScriptWindow might even be on another screen to the one the action was invoked from. Can we pass in a parent instead?

Reply:
Pass this in in https://github.com/GafferHQ/gaffer/pull/3560/commits/697961f19085319ce5c368c6b7b653ed8b4d3e6d so it appears on the right monitor.